### PR TITLE
Fix issue where slurmdbd doesn't restart due to wrong permissions/ownership.

### PR DESCRIPTION
**Pull request by michael-balint**

### CODE REVIEW

**Review by ajdecon — APPROVED** [review]
LGTM

Approving so you can merge at will, but you may want to wait until you see CI pass

**Comment by dholt** [comment]
For completeness, I'm not seeing this failure on my test system:

```
ubuntu@polite-tapir:~$ ls -althr /etc/slurm/slurmdbd.conf
-rw------- 1 root root 774 Feb 23 22:12 /etc/slurm/slurmdbd.conf
ubuntu@polite-tapir:~$ systemctl status slurmdbd
● slurmdbd.service - Slurm DBD accounting daemon
     Loaded: loaded (/etc/systemd/system/slurmdbd.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2021-02-23 22:12:25 UTC; 6min ago
```

**Pull request merged**
